.github/workflows/codspeed.yml

```diff
@@ -41,19 +41,39 @@ jobs:
         if: steps.cache-py.outputs.cache-hit != 'true'
 
       - name: install rust stable
+        id: rust-toolchain
         uses: dtolnay/rust-toolchain@stable
+        with:
+          components: llvm-tools
 
       - name: cache rust
         uses: Swatinem/rust-cache@v2
 
-      - name: Install pydantic-core
+      - name: Compile pydantic-core for profiling
+        # build-prod does not include MiMalloc because it bypasses Maturin
+        run: |
+          pip install -e .
+          make build-prod
+          python -c 'import pydantic_core'
+        env:
+          CONST_RANDOM_SEED: 0 # Fix the compile time RNG seed
+          RUSTFLAGS: "-Cprofile-generate=${{ github.workspace }}/profdata"
+
+      - name: Gather pgo data
+        run: pytest
+
+      - name: Prepare merged pgo data
```

Comment:
Is the result any different if we use `pytest tests/benchmarks`?

Reply:
Pushed #686 to retry this.

Reply:
So https://github.com/pydantic/pydantic-core/pull/686#issuecomment-1602228310 suggests that just running the benchmarks leads to higher speed on the benchmarks, which is not at all a surprise. I am slightly wary it could be an overfit. I'll try to find some ways to discern what might be overall better.

Reply:
My guess is that the case of a given check passing is much more common in benchmarks and real life than the case of an error, but they error cases are quite common in tests.